pkg/windows/service_windows.go

```diff
@@ -79,5 +80,39 @@ func StartService() error {
 		}
 	}()
 
+	return true, nil
+}
+
+// MonitorProcessExit ensures that the kubelet, kube-proxy, calico-node, and containerd processes have stopped running.
```

Comment:
This is an interesting way to do this. I'm not a big fan of inline powershell and shelling out to do things that we should be able to do from native golang.

Can we not use a [WaitGroup](https://pkg.go.dev/sync#WaitGroup) when starting these processes through the PEBinaryExecutor, and just wait on that when shutting down?

Reply:
A bit of face palm moment on my side -  a WaitGroup is a much better idea. After looking into it a bit, I found `apimachinery/pkg/util/wait` which offers wait groups that utilize contexts. Spent some time Friday and this morning testing the changes out and everything works as desired without any inlined PowerShell. 

Let me know what you think. Also, I wanted to get your opinion on if it would be a good idea to add a timeout for the `WaitGroup.Wait` call. I can't think of a situation where one of these commands would continue to run after the context cancelation, but I want to avoid a situation where we might wait forever for the WaitGroup to complete.

Reply:
No I think this is good. If we're blocking on child process exit and want to be sure they get cleaned up, I think blocking shutdown indefinitely is appropriate.